addons/binding/org.openhab.binding.sensebox/src/main/java/org/openhab/binding/sensebox/handler/SenseBoxHandler.java

```diff
@@ -0,0 +1,373 @@
+
+                    publishDataForChannel(new ChannelUID(getThing().getUID(), CHANNEL_NAME));
+                    publishDataForChannel(new ChannelUID(getThing().getUID(), CHANNEL_LOCATION));
+                    publishDataForChannel(new ChannelUID(getThing().getUID(), CHANNEL_EXPOSURE));
+                    publishDataForChannel(new ChannelUID(getThing().getUID(), CHANNEL_API_URL));
+                    publishDataForChannel(new ChannelUID(getThing().getUID(), CHANNEL_MAP_URL));
+                    publishDataForChannel(new ChannelUID(getThing().getUID(), CHANNEL_IMAGE_URL));
+
+                    publishDataForChannel(new ChannelUID(getThing().getUID(), CHANNEL_UV_INTENSITY));
+                    publishDataForChannel(new ChannelUID(getThing().getUID(), CHANNEL_LUMINANCE));
+                    publishDataForChannel(new ChannelUID(getThing().getUID(), CHANNEL_PRESSURE));
+                    publishDataForChannel(new ChannelUID(getThing().getUID(), CHANNEL_HUMIDITY));
+                    publishDataForChannel(new ChannelUID(getThing().getUID(), CHANNEL_TEMPERATURE));
+
+                    publishDataForChannel(new ChannelUID(getThing().getUID(), CHANNEL_UV_INTENSITY_LR));
+                    publishDataForChannel(new ChannelUID(getThing().getUID(), CHANNEL_LUMINANCE_LR));
+                    publishDataForChannel(new ChannelUID(getThing().getUID(), CHANNEL_PRESSURE_LR));
+                    publishDataForChannel(new ChannelUID(getThing().getUID(), CHANNEL_HUMIDITY_LR));
+                    publishDataForChannel(new ChannelUID(getThing().getUID(), CHANNEL_TEMPERATURE_LR));
+
+                } catch (IOException e) {
+                    logger.warn("Exception occurred during fetching data: {}", e.getMessage(), e);
+                } catch (Exception e) {
+                    logger.debug("Exception occurred during execution: {}", e.getMessage(), e);
+                }
+            }
+        };
+
+        refreshJob = scheduler.scheduleWithFixedDelay(runnable, 0, thingConfiguration.getRefreshInterval(),
+                TimeUnit.SECONDS);
+    }
+
+    private SenseBoxData fetchData() throws IOException {
+        return CACHE.get(thingConfiguration.getSenseBoxId());
+    }
+
+    private final ExpiringCache<String, SenseBoxData> CACHE = new ExpiringCache<>(CACHE_EXPIRY,
+            new ExpiringCache.LoadAction<String, SenseBoxData>() {
+                @Override
+                public SenseBoxData load(String senseBoxId) throws IOException {
+                    return reallyFetchDataFromServer(senseBoxId);
+                }
+            });
+
+    private SenseBoxData reallyFetchDataFromServer(String senseBoxId) {
+        String query = SENSEMAP_API_URL_BASE + "/boxes/" + senseBoxId;
+
+        // the caching layer does not like null values
+        SenseBoxData result = new SenseBoxData();
+
+        try {
+            URL url = new URL(query);
+            URLConnection connection = url.openConnection();
```

Comment:
try-with-resources would require the URLConnection to be Autoclosable, right?  Do we have an example somewhere where this pattern is used on HTTP connections?

In any case, a close() on the InputStream is added, and this should close the connection too if I read the javadoc right.